### PR TITLE
Fixes many grammar issues and typos in JavaDoc and code comments

### PR DESCRIPTION
**Pull request by mawiesne**
Change
-
- fixes grammar issues in JavaDoc and code comments.
- fixes some typos found along the way.

### CODE REVIEW

**Comment by garydgregory** [comment]
Run a local build with 'mvn' before creating a PR. The build is broken.

**Comment by mawiesne** [comment]
> Run a local build with 'mvn' before creating a PR. This PR breaks the build.

```
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  01:29 min
[INFO] Finished at: 2023-01-13T20:05:50+01:00
[INFO] ------------------------------------------------------------------------
```
MacOS, OpenJDK 17.0.5

**Comment by mawiesne** [comment]
ic, checkstyle is not enabled by default, have to run it separately. will fix.

**Comment by garydgregory** [comment]
Look at the checks here on GitHub.

**Comment by mawiesne** [comment]
> Look at the checks here on GitHub. What is the exact command you ran?

did not expect `checkstyle:check` not be part of the regular build. solved it locally, np.

**Comment by mawiesne** [comment]
@garydgregory Could you re-run CI/8/Win, pls? Sth seems weird there:

```
Error:  Failures: 
Error:    TFTPServerPathTest.testReadOnly:55 Couldn't clear output location
Error:  Errors: 
Error:    TFTPServerPathTest.testWriteOutsideHome:116 � Bind Address already in use: Cannot bind
```

File: src/main/java/org/apache/commons/net/ftp/FTPSClient.java [hunk 1]
```diff
@@ -204,9 +204,9 @@ public FTPSClient(final String protocol, final boolean isImplicit) {
 
     /**
      * Because there are so many connect() methods, the _connectAction_() method is provided as a means of performing some action immediately after establishing
-     * a connection, rather than reimplementing all of the connect() methods.
+     * a connection, rather than reimplementing all the connect() methods.
      *
-     * @throws IOException If it throw by _connectAction_.
+     * @throws IOException If it throws by _connectAction_.
```
Comment (garydgregory) [inline src/main/java/org/apache/commons/net/ftp/FTPSClient.java:209]: Err...?

File: src/main/java/org/apache/commons/net/ftp/parser/MLSxEntryParser.java [hunk 2]
```diff
@@ -64,10 +64,10 @@ public class MLSxEntryParser extends FTPFileEntryParserImpl {
 
     private static final HashMap<String, Integer> TYPE_TO_INT = new HashMap<>();
     static {
-        TYPE_TO_INT.put("file", Integer.valueOf(FTPFile.FILE_TYPE));
```
Comment (garydgregory) [inline src/main/java/org/apache/commons/net/ftp/parser/MLSxEntryParser.java:67]: Don't change this code: leave boxing/unboxing alone.

File: src/main/java/org/apache/commons/net/ftp/parser/MLSxEntryParser.java [hunk 3]
```diff
@@ -234,7 +234,7 @@ public FTPFile parseFTPEntry(final String entry) {
                 if (intType == null) {
                     file.setType(FTPFile.UNKNOWN_TYPE);
                 } else {
-                    file.setType(intType.intValue());
+                    file.setType(intType);
```
Comment (garydgregory) [inline src/main/java/org/apache/commons/net/ftp/parser/MLSxEntryParser.java:237]: Please leave boxing/unboxing alone.

File: src/main/java/org/apache/commons/net/ftp/parser/MLSxEntryParser.java (hunk 2 above)
Comment (mawiesne) [reply src/main/java/org/apache/commons/net/ftp/parser/MLSxEntryParser.java:67]: Roger, will undo and push once more.

File: src/main/java/org/apache/commons/net/imap/IMAP.java [hunk 4]
```diff
@@ -374,7 +374,7 @@ public int sendCommand(final String command, final String args) throws IOExcepti
      * @param commandID The ID (tag) of the command.
      * @param command   The IMAP command to send.
      * @param args      The command arguments.
-     * @return The server reply code (either IMAPReply.OK, IMAPReply.NO or IMAPReply.BAD).
+     * @return The server reply code (either {@code IMAPReply.OK}, {@code IMAPReply.NO} or {@code IMAPReply.BAD}).
```
Comment (garydgregory) [inline src/main/java/org/apache/commons/net/imap/IMAP.java:377]: Probably use @link for these.

File: src/main/java/org/apache/commons/net/ftp/parser/MLSxEntryParser.java (hunk 3 above)
Comment (mawiesne) [reply src/main/java/org/apache/commons/net/ftp/parser/MLSxEntryParser.java:237]: ack

File: src/main/java/org/apache/commons/net/nntp/NNTPClient.java [hunk 5]
```diff
@@ -583,12 +583,12 @@ private void parseArticlePointer(final String reply, final ArticleInfo pointer)
...
      * <p>
      * You must not issue any commands to the NNTP server (i.e., call any (other methods) until you finish writing to the returned Writer instance and close it.
-     * The NNTP protocol uses the same stream for issuing commands as it does for returning results. Therefore the returned Writer actually writes directly to
+     * The NNTP protocol uses the same stream for issuing commands as it does for returning results. Therefore, the returned Writer actually writes directly to
      * the NNTP connection. After you close the writer, you can execute new commands. If you do not follow these requirements your program will not work
      * properly.
      * <p>
      * Different NNTP servers will require different header formats, but you can use the provided {@link org.apache.commons.net.nntp.SimpleNNTPHeader} class to
-     * construct the bare minimum acceptable header for most news readers. To construct more complicated headers you should refer to RFC 822. When the Java Mail
+     * construct the bare minimum acceptable header for most newsreaders. To construct more complicated headers you should refer to RFC 822. When the Java Mail
```
Comment (garydgregory) [inline src/main/java/org/apache/commons/net/nntp/NNTPClient.java:591]: "newsreaders" is not a word AFAIK

File: src/main/java/org/apache/commons/net/nntp/NNTPClient.java [hunk 6]
```diff
@@ -1283,7 +1283,7 @@ public boolean selectArticle(final long articleNumber, final ArticleInfo pointer
     /**
      * Same as <code> selectArticle(articleId, (ArticleInfo) null) </code>
      *
-     * @param articleId the article Id
+     * @param articleId the article's identifier
```
Comment (garydgregory) [inline src/main/java/org/apache/commons/net/nntp/NNTPClient.java:1286]: Superfluous change IMO, this change makes the text wordier, it was clearer to me previously.

File: src/main/java/org/apache/commons/net/nntp/NewGroupsOrNewsQuery.java [hunk 7]
```diff
@@ -246,7 +246,7 @@ public boolean isGMT() {
      * </pre>
      * <p>
      *
-     * @param newsgroup The newsgroup to add to the list of groups to be checked for new news, but which should be omitted from the search for new news..
+     * @param newsgroup The newsgroup to add to the list of groups to be checked for new news, but which should be omitted from the search for recent news.
```
Comment (garydgregory) [inline src/main/java/org/apache/commons/net/nntp/NewGroupsOrNewsQuery.java:249]: Why "fresh" before and "recent" here? Using "recent" for both would be more consistent IMO.

File: src/main/java/org/apache/commons/net/nntp/Threader.java [hunk 8]
```diff
@@ -56,29 +55,29 @@ private void buildContainer(final Threadable threadable, final HashMap<String, T
             }
         }
 
-        // No container exists for that message Id. Create one and insert it into the hash table.
+        // No container exists for that message identifier. Create one and insert it into the hash table.
```
Comment (garydgregory) [inline src/main/java/org/apache/commons/net/nntp/Threader.java:58]: No need to make the code comments wordier.

File: src/main/java/org/apache/commons/net/nntp/NNTPClient.java (hunk 5 above)
Comment (mawiesne) [reply src/main/java/org/apache/commons/net/nntp/NNTPClient.java:591]: What about this: https://www.newsgroupreviews.com/newsgroup-readers.html, see title? Saw it more recently in this form.

Comment (mawiesne) [reply src/main/java/org/apache/commons/net/nntp/NNTPClient.java:591]: See also: https://en.wikipedia.org/wiki/List_of_Usenet_newsreaders

Comment (garydgregory) [reply src/main/java/org/apache/commons/net/nntp/NNTPClient.java:591]: Gross, but ok I guess.

File: src/main/java/org/apache/commons/net/ftp/FTPSClient.java [hunk 9]
```diff
@@ -204,7 +204,7 @@ public FTPSClient(final String protocol, final boolean isImplicit) {
 
     /**
      * Because there are so many connect() methods, the _connectAction_() method is provided as a means of performing some action immediately after establishing
-     * a connection, rather than reimplementing all of the connect() methods.
+     * a connection, rather than reimplementing all the connect() methods.
      *
      * @throws IOException If it throw by _connectAction_.
```
Comment (garydgregory) [inline src/main/java/org/apache/commons/net/ftp/FTPSClient.java:209]: Still not correct.

Comment (mawiesne) [reply src/main/java/org/apache/commons/net/ftp/FTPSClient.java:209]: That's what was there before. 
Any suggestion how that should be formulated?

Comment (garydgregory) [reply src/main/java/org/apache/commons/net/ftp/FTPSClient.java:209]: > That's what was there before. Any suggestion how that should be formulated?

Use a JDK method as a template

File: src/main/java/org/apache/commons/net/ftp/parser/UnixFTPEntryParser.java [hunk 10]
```diff
@@ -110,7 +110,7 @@ public class UnixFTPEntryParser extends ConfigurableFTPFileEntryParserImpl {
     /**
      * The default constructor for a UnixFTPEntryParser object.
      *
-     * @throws IllegalArgumentException Thrown if the regular expression is unparseable. Should not be seen under normal conditions. It it is seen, this is a
+     * @throws IllegalArgumentException Thrown if the regular expression is unparseable. Should not be seen under normal conditions. If it is seen, this is a
```
Comment (garydgregory) [inline src/main/java/org/apache/commons/net/ftp/parser/UnixFTPEntryParser.java:113]: Might this be better: "If it is seen" -> "If it is thrown"

Comment (mawiesne) [reply src/main/java/org/apache/commons/net/ftp/parser/UnixFTPEntryParser.java:113]: Would apply for many other occurrences as well. Could be a separate PR, wdyt?

**Comment by garydgregory** [comment]
Up to you. When you are done with edits here, I will review the PR again (hopefully for the last time).

File: src/main/java/org/apache/commons/net/ftp/parser/UnixFTPEntryParser.java (hunk 10 above)
Comment (garydgregory) [reply src/main/java/org/apache/commons/net/ftp/parser/UnixFTPEntryParser.java:113]: Up to you. When you are done with edits here, I will review the PR again (hopefully for the last time).

Comment (garydgregory) [reply src/main/java/org/apache/commons/net/ftp/parser/UnixFTPEntryParser.java:113]: Hm, then if I was confused by what "it" refers to, can we replace the "it" with the actual subject?

File: src/main/java/org/apache/commons/net/nntp/NNTPClient.java [hunk 11]
```diff
@@ -268,9 +268,9 @@ public Iterable<Article> iterateArticleInfo(final long lowArticleNumber, final l
     }
 
     /**
-     * List all new articles added to the NNTP server since a particular date subject to the conditions of the specified query. If no new new news is found, no
-     * entries will be returned. This uses the "NEWNEWS" command. You must add at least one newsgroup to the query, else the command will fail. Each String
-     * which is returned is a unique message identifier including the enclosing &lt; and &gt;.
+     * List all new articles added to the NNTP server since a particular date subject to the conditions of the specified query. If no recent news is found,
```
[the remark this answers is not in the document]
Comment (mawiesne) [reply src/main/java/org/apache/commons/net/nntp/NNTPClient.java:271]: Convinced. I'll revert the three occurrences of that text fragment with the next commit.

File: src/main/java/org/apache/commons/net/ftp/FTPSClient.java (hunk 9 above)
Comment (mawiesne) [reply src/main/java/org/apache/commons/net/ftp/FTPSClient.java:209]: I've changed it according to the style found/used in `FTPSClient`.
To be found in next commit together with changes proposed by @sebbASF.

File: src/main/java/org/apache/commons/net/ftp/parser/UnixFTPEntryParser.java (hunk 10 above)
Comment (mawiesne) [reply src/main/java/org/apache/commons/net/ftp/parser/UnixFTPEntryParser.java:113]: I replaced "it" with "this exception" and left "seen" as is.

**Comment by garydgregory** [comment]
@sebbASF Have all your concerns been addressed?

**Comment by mawiesne** [comment]
@sebbASF  My latest commit contained JavaDoc only changes. However,

```
Error:  Errors: 
Error:    TFTPServerPathTest.testWriteOutsideHome:116 � Bind Address already in use: Cannot bind
```

on Java CI / 8, windows seems to toggle under certain circumstances. I guess something is weird for this CI action and/or test setup / exec environment. 

Nevertheless, could you re-run this particular action, pls?

**Comment by garydgregory** [comment]
TY @mawiesne 
TY @sebbASF

**Pull request merged**